frontend/javascripts/oxalis/model/bucket_data_handling/layer_rendering_manager.ts

```diff
@@ -103,8 +103,9 @@ export function getGlobalLayerIndexForLayerName(
 ): number {
   const sanitizer = optSanitizer || _.identity;
   const dataset = Store.getState().dataset;
-  const allLayers = [...getColorLayers(dataset), ...getSegmentationLayers(dataset)];
-  const layerIndex = allLayers.findIndex((layer) => sanitizer(layer.name) === layerName);
```

Comment:
this old code is similar to how the shader code constructs the layer order. however, we need the indices from `dataset.dataSource.dataLayers` because that ordering is used via `Object.values` in multiple places to set up uniforms (e.g., for passing the active mag indices and the resolutions of the layers). for datasets where all layers are homogenous (or where the ordering matched by chance), this was not an issue.